torch/testing/_internal/common_methods_invocations.py

```diff
@@ -578,22 +578,85 @@ def sample_inputs_linalg_vector_norm(op_info, device, dtype, requires_grad):
 
     return inputs
 
+def sample_inputs_mm(op_info, device, dtype, requires_grad):
+    args_list = (
+        ((S, M), (M, S)),
+    )
+    inputs = list(SampleInput(make_tensor(first_shape, device, dtype,
+                                          requires_grad=requires_grad),
+                              args=(make_tensor(second_shape, device, dtype,
+                                    requires_grad=requires_grad),))
+                  for first_shape, second_shape in args_list)
+    return inputs
+
+def sample_inputs_add(op_info, device, dtype, requires_grad):
+    inputs = []
+    args_list = (
```

Comment:
cc @kshitij12345 

What's interesting here is that the lhs can require broadcasting to the rhs. How does this pass test_variant_consistency_eager, where the inplace version of the operator is compared to the functional version?

@kshitij12345 is working a PR to support this case now, by the way.

Reply:
lhs broadcasted sample working for inplace variant goes back to our conversation at https://github.com/pytorch/pytorch/pull/54945#discussion_r607039433

Reply:
That's what I thought. We don't have an issue for that, do we? Would you like to submit a PR fixing it now @kshitij12345? If not, I can take a look at it, too. It seems super high priority.

Reply:
It is taken care of in https://github.com/pytorch/pytorch/pull/53014.

I had drafted a seperate PR for it https://github.com/pytorch/pytorch/pull/55300 (but closed it in favor of #53014)

If it is high-priority, then I can reopen #55300 and get it merged first.
Let me know what you think.

Reply:
@kshitij1234, post #53014, would sample inputs that cause an op's inplace variant to resize the base tensor have some kwargs?

Reply:
This PR will update the recommended signature for sample_inputs to `sample_inputs_func(op_info, device, dtype, requires_grad, **kwargs)`. 
With that kwargs can now contain various information from tests if necessary.
For this particular case it will have a flag called `for_inplace_variant` based on which you can update the sample that the test returns.
Eg.
https://github.com/pytorch/pytorch/blob/55076c7a4ac8eb95ed97961b529f20ac3f9f49ef/torch/testing/_internal/common_methods_invocations.py#L1696-L1732

Hope that makes sense. Let me know if you have any questions